### PR TITLE
Relay

### PR DESCRIPTION
**Pull request by chux0519**
- uprade libp2p version (with blacklist and some bug fixes)
- add API to disconnect a specific connection

### CODE REVIEW

File: internal/pkg/conn/p2p/relayfilter.go [hunk 1]
```diff
@@ -38,11 +38,17 @@ func (rf *QuorumRelayFilter) AllowConnect(src peer.ID, srcAddr ma.Multiaddr, des
 	}
 
 	// check whether the remote peer is in the blacklist of server peer
+	// should check both side, cause connection could be bio connection
 	inBlacklist, err := handlers.CheckBlacklist(rf.db, dest.String(), src.String())
 	if err != nil {
 		// db error, we abort connect by now
 		return false
 	}
+	inBlacklistRev, err := handlers.CheckBlacklist(rf.db, dest.String(), src.String())
```
Comment (huo-ju) [inline internal/pkg/conn/p2p/relayfilter.go:47]: I can't find difference with line 42?   Both `CheckBlacklist` funcs have same args.

File: pkg/autorelay/api/blacklist.go [hunk 2]
```diff
@@ -33,6 +34,19 @@ func (h *RelayServerHandler) AddBlacklist(c echo.Context) (err error) {
...
+	if relay != nil {
+		from, err := peer.Decode(param.FromPeer)
+		if err != nil {
+			return rumerrors.NewBadRequestError(err.Error())
+		}
+		to, err := peer.Decode(param.ToPeer)
+		if err != nil {
+			return rumerrors.NewBadRequestError(err.Error())
+		}
+		relay.DisconnectByPeerID(from, to)
```
Comment (huo-ju) [inline pkg/autorelay/api/blacklist.go:47]: The libp2p connection It's a bidirectional connection, and is it need call an extra relay.DisconnectByPeerID(to, from)?

Comment (chux0519) [reply pkg/autorelay/api/blacklist.go:47]: yeah true

Comment (chux0519) [reply pkg/autorelay/api/blacklist.go:47]: nah, either side is closed, the other side will be closed too, no need to call it twice. that means, `DisconnectByPeerID(from, to)` is the same as `DisconnectByPeerID(to, from)` , just affects from which node to disconnect

but things are different during the connection establishment phase, a connection can be requested by both two peers, so if we want to blacklist a connection, we have to check both direction, so we call `CheckBlacklist` twice [here](https://github.com/rumsystem/quorum/pull/193/files#diff-7f67a0b91ba6efff251c250d37614b593f2e461ef140eee7001707d3d8bec914R47)

**Pull request merged**